sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/ReplaceNullWithFalseInPredicate.scala

```diff
@@ -94,6 +94,7 @@ object ReplaceNullWithFalseInPredicate extends Rule[LogicalPlan] {
         replaceNullWithFalse(cond) -> replaceNullWithFalse(value)
       }
       val newElseValue = cw.elseValue.map(replaceNullWithFalse)
```

Comment:
How about `val newElseValue = replaceNullWithFalse(cw.elseValue.getOrElse(null_literal))`? The rule `SimplifyConditionals` will help us to remove redundant branches.